oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -208,6 +210,27 @@ private ImpersonatedCredentials initializeImpersonatedCredentials() {
         .build();
   }
 
+  @Override
+  public void getRequestMetadata(
+      URI uri, Executor executor, final RequestMetadataCallback callback) {
+    super.getRequestMetadata(
+        uri,
+        executor,
+        new RequestMetadataCallback() {
+          @Override
+          public void onSuccess(Map<String, List<String>> metadata) {
+            metadata = addQuotaProjectIdToRequestMetadata(quotaProjectId, metadata);
```

Comment:
do we need to do this here? the quotaProjectId is being added in the override below, which seems like is always called

Reply:
This doesn't call the override below. It calls `super.getRequestMetadata`

Reply:
But I did just notice that I had a typo & invoked super.getRequestMetadata twice.

Reply:
ah calling super.getRequestMetadata used to call the override below

Reply:
yea, I dont love the parallel logic paths, and tried to call it out in the comment above, but I dont have a better approach. If you have ideas, lets keep the discussion in https://github.com/googleapis/google-auth-library-java/pull/646/files#r625406829